### ISSUE TITLE
Support batching sequence labels pairs for zero shot classification pipeline

### ISSUE THREAD

**Issue by ABasharEter** [huggingface/transformers#29793]
### Feature request

Currently zero shot classification pipeline dose not support specifying different set of labels for each sample input when batching multiple sequences of input. We want support input of a pair of sequence and its labels so that we can run prediction on them in batch.

Following is an example of what sample input we want to specify:

```
samples = [
("Iphone 15 pro max", ["mobiles", "appliances", "toys"]),
("Iphone 15 pro max", ["Samsung", "Apple"]),
("galaxy s21 ultra", ["mobiles", "appliances", "toys"]),
("galaxy s21 ultra", ["Samsung", "Apple"])
]
```

### Motivation

By grouping labels with sequences pairs we can batch different type of labels for each sample and by batching it runs on gpu will be faster.

### Your contribution

I will work on this feature and create a pr for it then update this issue. Making the issue now just to make sure that no one is working on this while I am doing it.

**Comment by qgallouedec** [comment]
Hey @ABasharEter, sorry for the very slow reply, we're catching up on the backlog.

Interest in this never picked up (no other requests since you filed it), and #29866 would have added a second public pipeline task that we'd be stuck supporting, so we're going to close it here.

You don't need a new pipeline for the batching win though. Zero-shot is just NLI pairs plus a softmax over the entailment logits, so you can build the pairs yourself with per-sample label sets and send them through `text-classification` in one batched call:

```python
pairs = [{"text": seq, "text_pair": f"This example is {lab}."}
         for seq, labels in samples for lab in labels]
raw = pipe(pairs, top_k=None, function_to_apply="none", batch_size=32)
```

Then take the entailment score of each pair and normalize per sample, as the pipeline does in

https://github.com/huggingface/transformers/blob/8445b13cd24961e47f25a649fb113580f71a8d11/src/transformers/pipelines/zero_shot_classification.py#L252-L254
